eng/scripts/Automation-Sdk-UpdateChangelog.ps1

```diff
@@ -0,0 +1,389 @@
+function Get-LatestReleasedStableVersion {
+    param(
+        [string]$GroupId,
+        [string]$ArtifactId
+    )
+    
+    $groupPath = $GroupId -replace '\.', '/'
+    $metadataUrl = "https://repo1.maven.org/maven2/$groupPath/$ArtifactId/maven-metadata.xml"
+    
+    try {
+        $response = Invoke-WebRequest -Uri $metadataUrl -UseBasicParsing -ErrorAction Stop
+        [xml]$metadata = $response.Content
+        
+        # Get all versions and reverse to get latest first
+        $versions = $metadata.metadata.versioning.versions.version
+        if ($versions -is [string]) {
+            $versions = @($versions)
+        }
+        [array]::Reverse($versions)
+        
+        # Find the latest stable version (non-beta), otherwise use latest
+        $latestVersion = $metadata.metadata.versioning.latest
+        foreach ($version in $versions) {
+            if ($version -notmatch "-beta") {
+                $latestVersion = $version
+                break
+            }
+        }
+        
+        return $latestVersion
+    }
+    catch {
+        Write-Warning "Could not retrieve metadata from Maven Central. Package may not be released yet."
+        return $null
+    }
+}
+
+# Function to download JAR file from Maven Central
+function Get-MavenJar {
+    param(
+        [string]$GroupId,
+        [string]$ArtifactId,
+        [string]$Version,
+        [string]$OutputPath
+    )
+    
+    $groupPath = $GroupId -replace '\.', '/'
+    $jarFileName = "$ArtifactId-$Version.jar"
+    $jarUrl = "https://repo1.maven.org/maven2/$groupPath/$ArtifactId/$Version/$jarFileName"
+    $jarPath = Join-Path $OutputPath $jarFileName
+    
+    Write-Host "Downloading JAR from: $jarUrl"
+    Invoke-WebRequest -Uri $jarUrl -OutFile $jarPath -UseBasicParsing
+    
+    return $jarPath
+}
+
+# Function to find the built JAR file in the target directory
+function Get-BuiltJarPath {
+    param(
+        [string]$PackagePath,
+        [string]$ArtifactId
+    )
+    
+    $targetPath = Join-Path $PackagePath "target"
+    if (-not (Test-Path $targetPath)) {
+        throw "Target directory not found. Please build the package first."
+    }
+    
+    # Look for JAR files matching the artifact name (excluding sources and javadoc)
+    $jarFiles = Get-ChildItem -Path $targetPath -Filter "$ArtifactId-*.jar" | 
+                Where-Object { $_.Name -notmatch "-sources\.jar$" -and $_.Name -notmatch "-javadoc\.jar$" }
+    
+    if ($jarFiles.Count -eq 0) {
+        throw "No JAR file found in target directory"
+    }
+    
+    # Return the first matching JAR (should be the main artifact)
+    return $jarFiles[0].FullName
+}
+
+# Function to run the changelog generation tool
+function Invoke-ChangelogGeneration {
+    param(
+        [string]$SdkRepoPath,
+        [string]$OldJarPath,
+        [string]$NewJarPath
+    )
+    
+    $changelogPomPath = Join-Path $SdkRepoPath "eng\automation\changelog\pom.xml"
```

Comment:
The path separator is Windows style. Will it work on Linux (maybe PS handles it, but better to have reference or test)?

Reply:
updated